stubs/backend-proxy.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const express = require("express");
+const http = require("http");
+const httpProxy = require('http-proxy');
+
+const assemblyProxy = httpProxy.createProxyServer({
+  host: 'http://localhost:4631'
+});
+
+const documentsProxy = httpProxy.createProxyServer({
+  host: 'http://localhost:4603'
+});
+
+const idamToken = require('../bin/IdamHelper').getIdamToken();
+const s2sToken = require('../bin/S2SHelper').getS2sToken();
+
+
+console.log('here\'s the idam token in the proxy: ' + idamToken);
+console.log('here\'s the s2s token in the proxy: ' + s2sToken);
+const app = express();
+
+// app.use(bodyParser.json());
+
+app.use((req, res, next) => {
+  res.header('Access-Control-Allow-Origin', '*');
+  res.header('Access-Control-Allow-Headers', 'Origin, X-Requested-With, Content-Type, Accept');
+  next();
+});
+
+app.use('/api', async (req, res, next) => {
+
+  req.headers['Authorization'] = await idamToken;
```

Comment:
This will await the same promise every time, even after it's resolved. Better to resolve it before starting the express service and then continuing to use it.

Reply:
moved out